app/components/galoy-text-input/galoy-text-input.tsx

```diff
@@ -0,0 +1,110 @@
+import { palette } from "@app/theme"
+import React, { useEffect, useState } from "react"
+import { TextInput, View } from "react-native"
+import { Text } from "react-native-elements"
+import EStyleSheet from "react-native-extended-stylesheet"
+
+type GaloyTextInputProps = {
+  placeholder?: string
+  append?: string
+  prepend?: string
+  validation?: [() => boolean]
+  label: string
+  onChangeText: (text: string) => void
+  labelStyle?: EStyleSheet.AnyObject
+}
+
+const styles = EStyleSheet.create({
+  container: {
+    flexDirection: "column",
+  },
+  labelContainer: {
+    flex: 1,
+  },
+  inputContainer: {
+    flex: 1,
+  },
+  label: {
+    color: palette.inputLabel,
+    fontSize: 18,
+    lineHeight: 24,
+    fontFamily: "Roboto",
+    fontWeight: "500",
+  },
+  input: {
+    backgroundColor: palette.inputBackground,
+    borderRadius: 8,
+    paddingVertical: 8,
+    paddingHorizontal: 12,
+  },
+  appendTextStyle: {
+    color: palette.secondaryText,
+  },
+  prependTextStyle: {
+    color: palette.secondaryText,
+  },
+})
+
+export const GaloyTextInput = ({
+  placeholder,
+  label,
+  labelStyle,
+  append = "",
+  prepend = "",
+  onChangeText,
+}: GaloyTextInputProps) => {
+  const [value, setValue] = React.useState("")
+  const [cleanValue, setCleanValue] = useState("")
+
+  useEffect(() => {
```

Comment:
I think by using ```useEffect``` in this component you are introducing unnecessary rerenders. For example user calls setValue which triggers re-render with new value which triggers useEffect in line 68 which sets cleanedValue which causes a rerender which finally triggers useEffect in line 80. These could all happen in the same callback suchas:
```
handleChangeText( text) => {
    // filter text to cleanValue
    setCleanValue(cleanValue)
    onChangeText(cleanValue)
}
```